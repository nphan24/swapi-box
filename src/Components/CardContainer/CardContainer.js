import { Switch, Route } from 'react-router-dom';
import React from 'react';
import People from '../People/People';
import Planets from '../Planets/Planets';
import Vehicles from '../Vehicles/Vehicles';
import Favorites from '../Favorites/Favorites';
import PropTypes from 'prop-types';

const CardContainer = ({ people, planets, vehicles, setFavorites, favorites }) => {
  
  return (
    <div>
      <Switch>
        <Route 
          exact path='/people' 
          component={() => <People info={people} setFavorites={setFavorites}/>}
        />
        <Route
          exact path='/planets'
          component={() => <Planets info={planets} setFavorites={setFavorites}/>}
        />
        <Route
          exact path='/vehicles'
          component={() => <Vehicles info={vehicles} setFavorites={setFavorites} />}
        />
        <Route
          exact path='/favorites'
          component={() => <Favorites info={favorites} setFavorites={setFavorites} />}
        />
      </Switch>
    </div>
  );
};

CardContainer.propTypes = {
  people: PropTypes.object,
  planets: PropTypes.object,
  vehicles: PropTypes.object,
  favorites: PropTypes.object,
  setFavorites: PropTypes.func
};

export default CardContainer;