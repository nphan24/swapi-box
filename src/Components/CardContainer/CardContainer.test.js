import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer', () => {
  let wrapper;
  let mockFunction;
  let mockArray;

  beforeEach(() => {

    mockArray = [{
      name: 'Cat',
      homeworld: 'Tatooine',
      population: '5'
    }];

    mockFunction = jest.fn();
    wrapper = shallow(<CardContainer info={mockArray} setFavorites={mockFunction} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});