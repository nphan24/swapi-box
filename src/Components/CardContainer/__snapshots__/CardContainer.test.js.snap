// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    info={
        Array [
            Object {
              "homeworld": "Tatooine",
              "name": "Cat",
              "population": "5",
            },
          ]
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                component={[Function]}
                path="/people"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/planets"
        />
        <Route
                component={[Function]}
                path="/vehicles"
        />
        <Route
                component={[Function]}
                exact={true}
                path="/favorites"
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            path="/people"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/planets"
/>,
          <Route
            component={[Function]}
            path="/vehicles"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/planets",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/vehicles",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    component={[Function]}
                    path="/people"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/planets"
          />
          <Route
                    component={[Function]}
                    path="/vehicles"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/favorites"
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/people"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/planets"
/>,
            <Route
              component={[Function]}
              path="/vehicles"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/favorites"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/people",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/planets",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/vehicles",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
